### PR TITLE
448: fix link in files table

### LINKED ISSUE #448: View Dataset Page: dataset version not included in files table link to File Page
When I dataset has multiple versions and includes a file, the link from the files table to the File Page should include the dataset version.  The datasetVersion param is missing from this link, so it always goes to the :latest version of the File.

To reproduce:

1. Create a dataset that has a file.  
2. Publish the dataset, which creates version 1.0
3. Edit the metadata of the file in the dataset, which creates a DRAFT
4. Go to the Dataset Page, with '&version=1.0' in the URL
5. From the Dataset Page, click on the file link in the files table.
6. This should go to the File Page with '&datasetVersion=1.0' in the URL, to show the 1.0 version of the file.  Instead, the datasetVersion param is missing, and it shows the :latest, which is DRAFT.

### PR DESCRIPTION
**Pull request by ekraffmiller**
**What this PR does / why we need it**:
Adds versionNumber to the FilePreview object so it can be used in the link to the File Page from the files table in Dataset Page

**Which issue(s) this PR closes**:

Closes #448 

**Special notes for your reviewer**:
I made datasetVersionNumber optional in FilePreview because it is only needed in the files table of the Dataset Page, it's not needed in the FileCard (the other component using FilePreview)

**Suggestions on how to test this**:
Create a dataset that has a file.  Edit Dataset and publish the dataset multiple times to create multiple versions. Go to an older version of the Dataset Page by using the version param.  In the files table of the Dataset Page, click on the file link.  It should go to the File Page showing the same version as the dataset.

**Does this PR introduce a user interface change? If mockups are available, please link/include them here**:
No
**Is there a release notes update needed for this change?**:
No
**Additional documentation**:

### CODE REVIEW

**Review by GermanSaracca — CHANGES_REQUESTED** [review]
Nice work, just a comment about the error throwing.

File: src/sections/dataset/dataset-files/files-table/file-info/file-info-cell/file-info-data/FileTitle.tsx [hunk 1]
```diff
@@ -1,14 +1,18 @@
...
+  datasetVersionNumber: DatasetVersionNumber
 }
 
-export function FileTitle({ id, name }: FileTitleProps) {
+export function FileTitle({ id, name, datasetVersionNumber }: FileTitleProps) {
   return (
-    <LinkToPage page={Route.FILES} searchParams={{ id: id.toString() }}>
+    <LinkToPage
+      page={Route.FILES}
+      searchParams={{ id: id.toString(), datasetVersion: datasetVersionNumber.toSearchParam() }}>
```
[the remark this answers is not in the document]
Comment (ekraffmiller) [reply src/sections/dataset/dataset-files/files-table/file-info/file-info-cell/file-info-data/FileTitle.tsx:15]: The filePreview param should always have datasetVersion...would it be better to create modified type, like this?
```
export interface FilePreviewWithDatasetVersion extends Omit<FilePreview, 'datasetVersion'> {
  datasetVersion: DatasetVersionNumber
}
```

**Comment by ekraffmiller** [comment]
@GermanSaracca I think you are right that changing the interface introduced more changes than was necessary for that small piece of logic, so I added a check for null as suggested 👍

File: src/sections/dataset/dataset-files/files-table/file-info/file-info-cell/FileInfoCell.tsx [hunk 2]
```diff
@@ -13,13 +13,17 @@ import { FileDescription } from './file-info-data/FileDescription'
 import { FileLabels } from '../../../../../file/file-labels/FileLabels'
 
 export function FileInfoCell({ file }: { file: FilePreview }) {
+  if (!file.datasetVersionNumber) {
+    throw new Error('FilePreview object must contain datasetVersionNumber')
```
[the remark this answers is not in the document]
Comment (ekraffmiller) [reply src/sections/dataset/dataset-files/files-table/file-info/file-info-cell/FileInfoCell.tsx]: Agreed, I added a check for undefined, which returns null

**Review by g-saracca — DISMISSED** [review]
Looks good, approving!

**Comment by GPortas** [comment]
For when you are back, can you please resolve the conflicts? thanks! @ekraffmiller

**Comment by g-saracca** [comment]
@GPortas I have resolved merge conflicts, can you take a look if everything still looks fine?

**Review by GPortas — APPROVED** [review]
Works good.

https://github.com/user-attachments/assets/019fdc5e-069b-4859-809f-92320e660ad4

**Pull request merged**